addons/misra.py

```diff
@@ -255,6 +255,36 @@ def getForLoopExpressions(forToken):
             lpar.astOperand2.astOperand2.astOperand1,
             lpar.astOperand2.astOperand2.astOperand2]
 
+def getWhileLoopExpressions(whileToken):
+    if not whileToken or whileToken.str != 'while':
+        return None
+    lpar = whileToken.next
+    if not lpar or lpar.str != '(':
+        return None
+    if not lpar.astOperand2.astOperand1:
```

Comment:
I think you should check what lpar.astOperand2 is. if the code is:

    while (a<100 || b > 100)

then the `counter` will be `<`